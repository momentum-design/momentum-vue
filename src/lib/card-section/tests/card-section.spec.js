import { shallowMount } from '@vue/test-utils';
import CardSection from '../index.vue';

describe('CardSection', () => {
  it('should match snapshot', () => {
    const container = shallowMount(CardSection, {
      attrs: {
        id: 'test'
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one CardSection', () => {
    const container = shallowMount(CardSection, {
      attrs: {
        id: 'test'
      }
    });

    expect(container.findAll('.md-card-section').length).toEqual(1);
  });

  it('should render one full CardSection', () => {
    const container = shallowMount(CardSection, {
      propsData: {
        full: true
      }
    });

    expect(container.findAll('.md-card-section--full').length).toEqual(1);
  });

  it('should render a CardSection with the given classes', () => {
    const container = shallowMount(CardSection, {
      attrs: {
        class: 'testing-class'
      }
    });

    expect(container.classes('testing-class')).toEqual(true);
  });

  it('should render children', () => {
    const container = shallowMount(CardSection, {
      slots: {
        default: `<div class="testingforMB" />`
      }
    });

    expect(container.findAll('.testingforMB').length).toEqual(1);
  });
});

