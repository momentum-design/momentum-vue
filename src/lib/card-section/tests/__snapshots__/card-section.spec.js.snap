// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSection should match snapshot 1`] = `<div class="md-card-section" id="test"></div>`;
